app/graphql/sources/linked_to_editions_source.rb

```diff
@@ -0,0 +1,27 @@
+module Sources
+  class LinkedToEditionsSource < GraphQL::Dataloader::Source
+    def initialize(parent_object:)
+      @object = parent_object
+      @content_store = parent_object.content_store.to_sym
+    end
+
+    def fetch(link_types)
+      all_links = @object.document.link_set_links
+        .includes(target_documents: @content_store) # content_store is :live or :draft (a Document has_one Edition by that name)
+        .where(link_type: link_types)
+        .where(target_documents: { locale: "en" })
+
+      link_types_map = link_types.index_with { [] }
+
+      all_links.each_with_object(link_types_map) { |link, hash|
+        hash[link.link_type].concat(editions_for_link(link))
+      }.values
+    end
+
+  private
+
+    def editions_for_link(link)
+      link.target_documents.map { |document| document.send(@content_store) }
```

Comment:
Is `send` a safe thing to do based on a consumer's input?  I'm thinking a query could contain any value for `content_store` which we'll use to execute a method of the same name on the document.  Would it be better to explicitly write out `where(content_store: @content_store` here?

Reply:
That's smart thinking

Reply:
We could check that the provided value is a member of `Edition.content_stores.keys`, and specify a default one to fallback to.

But taking a closer look, it's only the database query for the root Edition that uses the value from the GraphQL query. The `content_store` of the links fields is taken from the Edition's database record. So I'm going with this being fine for now, could be made more robust in the longer term, in case of changes to how its used.